toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/popup/PopupViewTestActivity.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.test.popup
+
+import android.os.Bundle
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import com.esri.arcgisruntime.data.Feature
+import com.esri.arcgisruntime.geometry.GeometryType
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.toolkit.extension.logTag
+import com.esri.arcgisruntime.toolkit.popup.PopupView
+import com.esri.arcgisruntime.toolkit.popup.PopupViewModel
+import com.esri.arcgisruntime.toolkit.test.R
+import com.esri.arcgisruntime.toolkit.test.databinding.ActivityPopupBinding
+import com.esri.arcgisruntime.toolkit.util.observeEvent
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import kotlin.math.roundToInt
+
+/**
+ * Activity to show usages of [PopupView] and [PopupViewModel].
+ *
+ * - The activity's layout hosts the MapView and bottom sheet
+ * - Creates and owns the PopupViewModel for showing the Popup of an identified feature
+ * - It handles tap gestures on the MapView in order to identify a feature by overriding
+ * onSingleTapConfirmed on DefaultMapViewOnTouchListener
+ * - When a feature is identified, selects that feature in the MapView, initializes the
+ * PopupViewModel and shows the bottom sheet populated with PopupFragment.
+ * - The PopupFragment's layout comprises of edit layout which hosts the controls to activate/de-activate
+ * the edit-mode on the PopupView, delete feature, save and cancel edits. It also has the [PopupView] which
+ * shows the attributes of the identified feature in read-only mode upon initialization.
```

Comment:
I suggest to remove this last bullet point, as this is already explained in the class level doc of PopupFragment.

Reply:
done